MathPortal/MathPortal/Source/UserInterface/TabBarViewControllers/CreatedTasks/TaskViewController/TaskViewControllerTableViewCell.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  TaskViewControllerTableViewCell.swift
+//  MathPortal
+//
+//  Created by Petra Čačkov on 28/07/2019.
+//  Copyright © 2019 Petra Čačkov. All rights reserved.
+//
+
+import UIKit
+
+protocol TaskViewControllerTableViewCellDelegate: class {
+    func taskViewControllerTableViewCell(sender: TaskViewControllerTableViewCell, didDeleteCellAt location: CGPoint)
+    func taskViewControllerTableViewCell(sender: TaskViewControllerTableViewCell, didSelectCellAt location: CGPoint)
+}
+
+class TaskViewControllerTableViewCell: UITableViewCell {
+    
+    var equation: Equation?
+
+    @IBOutlet private var deleteView: UIView?
+    @IBOutlet private var equationView: UIView?
+    @IBOutlet var deleteLeadingConstraint: NSLayoutConstraint?
+    @IBOutlet var deleteButtonShadow: UIView?
+    @IBOutlet var deleteButtonView: UIView?
+    
+    weak var delegate: TaskViewControllerTableViewCellDelegate?
+    
+    private var panGestureRecognizer: UIPanGestureRecognizer?
+    private var tapGestureRecogniserDelete: UITapGestureRecognizer?
+    private var tapGestureRecogniserSelect: UITapGestureRecognizer?
+    private var cellLocation: CGPoint?
+    
+    private var deleteViewClosed: Bool = true {
+        didSet {
+            guard let deleteView = deleteView else { return }
+            deleteLeadingConstraint?.constant = deleteViewClosed ? 0 : deleteView.frame.width
+            UIView.animate(withDuration: 0.3) { self.layoutIfNeeded() }
+        }
+    }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        deleteViewClosed = true
+        if let deleteButtonShadow = deleteButtonShadow, let deleteButtonView = deleteButtonView {
+            deleteButtonView.layer.cornerRadius = deleteButtonView.frame.width/2
+            deleteButtonShadow.layer.cornerRadius = deleteButtonShadow.frame.width/2
+        }
+        setupGestureRecognisers()
+    }
+    
+    func setupCell(delegate: TaskViewControllerTableViewCellDelegate, equation: Equation) {
+        self.selectionStyle = .none
+        self.showsReorderControl = false
+        self.delegate = delegate
+        self.equation = equation
+        refreshEquation()
+    }
+    
+    private var currentView: UIView?
+    private func refreshEquation() {
+        guard let equation = equation else { return }
+        currentView?.removeFromSuperview()
+        if let view = equation.expression.generateView().view {
+            currentView = view
+            equationView?.addSubview(view)
+            view.frame.origin = CGPoint(x: 10, y: 5)
+        }
+    }
+
+    private func setupGestureRecognisers() {
+        panGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(swipe))
```

Comment:
Usually better to do it like 
```
panGestureRecognizer = {
    let recognizer = ...
    return recognizer
}()
```

This way you avoid optionals and you can easily copy-paste similar code without having naming collisions.